Week2/homework/Magic Query v.1.0.js

```diff
@@ -0,0 +1,141 @@
+const mysql = require('mysql');
+const rl = require('readline');
+
+var connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: '7566',
+  database: 'new_world',
+  multipleStatements: true
+});
+
+console.log('\n / Magic Query v.1.0 \\');
+console.log(`
+1- What is the capital of country X ? (Accept X from user)
+2- List all the languages spoken in the region Y (Accept Y from user)
+3- Find the number of cities in which language Z is spoken (Accept Z from user)
+4- Are there any countries that have A) Same official language ?
+5- Are there any countries that have B) Same region ?
+6- List all the continents with the number of languages spoken in each continent
+`);
+const prompts = rl.createInterface(process.stdin, process.stdout);
+
+const queryPrepareStatements = [
+  `PREPARE stmt FROM 'SELECT city.Name FROM country JOIN city ON country.code = city.CountryCode WHERE country.capital = city.ID AND country.Name = ?';`,
+  `PREPARE stmt FROM 'SELECT DISTINCT countrylanguage.Language FROM countrylanguage JOIN country ON country.code = countrylanguage.CountryCode WHERE country.Region = ?';`,
+  `PREPARE stmt FROM 'SELECT COUNT(city.Name) AS total FROM countrylanguage JOIN city ON city.CountryCode = countrylanguage.CountryCode WHERE countrylanguage.language = ?';`,
+  `SELECT Name, Language FROM countrylanguage INNER JOIN country ON countrylanguage.countrycode = country.code WHERE countrylanguage.IsOfficial = 'T' ORDER BY countrylanguage.language;`,
+  `SELECT Name, Region FROM country ORDER BY Region;`,
+  `SELECT continent, COUNT(language) AS Languages FROM (SELECT e1.Continent AS Continent, e2.Language AS Language FROM country AS e1 LEFT JOIN (SELECT * FROM countrylanguage WHERE isOfficial = 'T') AS e2 ON (e1.Code = e2.CountryCode)) AS l GROUP BY continent;`
+];
+
+function executeProgram() {
+  prompts.question('Which query do you want to do ==>> ', function(answer) {
+    if (answer < 7 && answer > 0) {
+      switch (answer) {
+        case '1':
+          QueryByInput('What is the capital of ==>> ', 0);
+          break;
+        case '2':
+          QueryByInput('List all the languages spoken in the region ==>> ', 1);
+          break;
+        case '3':
+          QueryByInput('Find the number of cities in which spoken language is ==>> ', 2);
+          break;
+        case '4':
```

Comment:
This command is returning a list of "undefined":
Afrikaans => undefined
Aimará => undefined
Aimará => undefined
Albaniana => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined
Arabic => undefined

Reply:
Has been fixed.